### ISSUE TITLE
SPPVALUE missing since 2020

### ISSUE THREAD

**Issue by andybeet** [NOAA-EDAB/comlandr#59]
since 2020 there are many thousand records (~43,000) that do not contain SPPVALUE. They are NA
This is the CAMS data period. 

Reached out to stockeff group. May need to reassess and reach out to CAMS

So it turns out there is a `data_source` column in CAMS. Both of CAMSAA_DLR and CAMSAA_VTR have missing SPPVALUE.
May need to read through the CAMS docs to see why the DLR data would not have SPPVALUE.

[CAMS docs](https://www.greateratlantic.fisheries.noaa.gov/ro/fso/reports/cams/cams_documentation/join-landings.html#data-sources)

**Comment by sgaichas** [comment]
Yes I see that in the `comland_raw` pull: looks like about 10% of post 2020 records are missing `SPPVALUE`; at least a couple appear to have 0.00000000 for `SPPLIVMT`

460876 records in `comland_raw |> dplyr::filter(YEAR>2019 & !is.na(SPPVALUE))`
42359 records in `comland_raw |> dplyr::filter(YEAR>2019 & is.na(SPPVALUE))`

**Comment by andybeet** [comment]
This still seems to be an issue
``` r
# pull data
a <- comlandr::get_comland_raw_data(channel)
# NAs post 2020
a$comland |>
  dplyr::filter(is.na(SPPVALUE), YEAR > 2020)
# total number of rows
a$comland |>
  dplyr::filter(YEAR > 2020) |> nrow()

# remove menhaden. total by year
a$comland |>
  dplyr::filter(is.na(SPPVALUE), YEAR > 2020, !(NESPP3 == 221)) |> 
  dplyr::group_by(YEAR) |> 
  dplyr::summarise(mt = sum(SPPLIVMT),
                   val = sum(SPPVALUE)) |> 
  dplyr::arrange(desc(mt))
# pull species codes
sp <- comlandr::get_species(channel)$data |> 
  dplyr::select(NESPP3,SPPNM) |> 
  dplyr::mutate(NESPP3 = as.numeric(NESPP3)) |> 
  dplyr::distinct()

# remove menhaden, total by species
a$comland |>
  dplyr::filter(is.na(SPPVALUE), YEAR > 2020, !(NESPP3 == 221)) |> 
  dplyr::group_by(NESPP3) |> 
  dplyr::summarise(mt = sum(SPPLIVMT),
                   val = sum(SPPVALUE),
                   .groups = "drop") |>  
  dplyr::arrange(desc(mt)) |> 
  dplyr::left_join(sp, by = "NESPP3")
```
## Meric Tons with no value

<img width="188" height="162" alt="Image" src="https://github.com/user-attachments/assets/5ea98b77-033e-4e14-9478-9b8a5c2f2ada" />

## Metric tons by species
<img width="396" height="237" alt="Image" src="https://github.com/user-attachments/assets/002dad66-3792-45dc-a756-72a8cf7e02a9" />

**Comment by andybeet** [comment]
For spiny_dogfish " I believe that our current approach drops any record lacking weight or length values. ": lead scientist
Additional info from c. legault: "Thankfully, StockEff keeps track of this for each stock in the Data fix-ups portion of the Survey module. For example, spiny dogfish uses the LW equation to derive expecatchwt from length frequencies while windowpane flounder treats the expcatchwt as zero (because for windowpane it is typically a few very small fish caught and the weight is below the precision used to store the weight in the database, so putting zero is reasonable).
